api/topic.go

```diff
@@ -65,6 +62,106 @@ func (api *API) getTopicPrivateHandler(w http.ResponseWriter, req *http.Request)
 	log.Event(ctx, "request successful", log.INFO, logdata) // NOTE: name of function is in logdata
 }
 
+// getSubtopicsPublicHandler is a handler that gets a topic by its id from MongoDB
+func (api *API) getSubtopicsPublicHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	vars := mux.Vars(req)
+	id := vars["id"]
+	logdata := log.Data{
+		"request_id": ctx.Value(dprequest.RequestIdKey),
+		"topic_id":   id,
+		"function":   "getSubtopicsPublicHandler",
+	}
+
+	// get topic from mongoDB by id
+	topic, err := api.dataStore.Backend.GetTopic(id)
+	if err != nil {
+		// no topic found to retrieve the subtopics from
+		handleError(ctx, w, err, logdata)
+		return
+	}
+
+	var result models.PublicSubtopics
+
+	if len(topic.Current.SubtopicIds) > 0 {
+		for _, item := range topic.Current.SubtopicIds {
+			// get sub topic from mongoDB by item
+			topic, err := api.dataStore.Backend.GetTopic(item)
+			if err != nil {
+				logdata["missing subtopic for id"] = item
+				log.Event(ctx, err.Error(), log.ERROR, logdata)
+				continue
+			}
+			result.PublicItems = append(result.PublicItems, topic.Current)
+			result.TotalCount++
+		}
+		if result.TotalCount == 0 {
+			handleError(ctx, w, apierrors.ErrInternalServer, logdata)
+			return
+		}
+	} else {
+		// no subtopics exist for the requested ID
+		handleError(ctx, w, apierrors.ErrNotFound, logdata)
+		return
+	}
+
+	// User is not authenticated and hence has only access to current sub document(s)
+	if err := WriteJSONBody(ctx, result, w, logdata); err != nil {
+		return
+	}
+	log.Event(ctx, "request successful", log.INFO, logdata) // NOTE: name of function is in logdata
```

Comment:
Consider removing this comment, I don't think it's needed.

Reply:
The comment IS needed because previous reviewers did not realise that logdata held info of use.